core/src/solve.jl

```diff
@@ -295,6 +299,93 @@ struct Parameters
     lookup::Dict{Int, Symbol}
 end
 
+"""
+Test for each node given its node type whether it has an allowed
+number of flow inneighbors and flow outneighbors
+"""
+function valid_flow_neighbor_amounts(p::Parameters)::Bool
+    (;
+        connectivity,
+        basin,
+        linear_resistance,
+        manning_resistance,
+        tabulated_rating_curve,
+        fractional_flow,
+        level_boundary,
+        flow_boundary,
+        pump,
+        terminal,
+    ) = p
+
+    (; graph_flow) = connectivity
+
+    errors = String[]
+
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, basin))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, linear_resistance))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, manning_resistance))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, tabulated_rating_curve))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, fractional_flow))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, level_boundary))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, flow_boundary))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, pump))
+    append!(errors, valid_flow_neighbor_amounts(graph_flow, terminal))
+
+    return if isempty(errors)
```

Comment:
Now, I don't really like implicit returns, and I guess this is explicit.... but I (personally) still don't like it.

I guess either do:

```julia
if isempty(errors)
    true
else
    @error join(errors, "\n")
    false
end
```

Or just nice and explicit as I prefer it ^^:

```julia
if isempty(errors)
    return true
else
    @error join(errors, "\n")
    return false
end
```